enhancements/ingress/haproxy-healthcheckinterval.md

```diff
@@ -0,0 +1,291 @@
+to the environment in
+[desiredRouterDeployment](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/controller/ingress/deployment.go#L206):
+```go
+// desiredRouterDeployment returns the desired router deployment.
+func desiredRouterDeployment(ci *operatorv1.IngressController, ingressControllerImage string, ingressConfig *configv1.Ingress, apiConfig *configv1.APIServer, networkConfig *configv1.Network, proxyNeeded bool, haveClientCAConfigmap bool, clientCAConfigmap *corev1.ConfigMap) (*appsv1.Deployment, error) {
+...
+if ci.Spec.TuningOptions.HealthCheckInterval != nil && ci.Spec.TuningOptions.HealthCheckInterval > 5*time.Second {
+    env = append(env, corev1.EnvVar{Name: "ROUTER_BACKEND_CHECK_TIMEOUT", 
+          Value: durationToHAProxyTimespec(ci.Spec.TuningOptions.HealthCheckInterval)})
+...
+}
+```
+The HAProxy template will not be modified.
+
+### Risks and Mitigations
+
+One risk of this proposal is that customers may inadvertently cause different problems with a long healthcheck
+interval.  For example, backend services that would normally be marked as DOWN would be seen as UP for a longer time
+period, causing more errors to be seen by their application users.  To mitigate this issue the healthcheck can be
+adjusted to a shorter interval, or set back to the default.
+
+Furthermore, a change to the health check `inter` duration can have an impact on the `timeout connect` for healthchecks,
+which is the maximum time to wait for a connection attempt to succeed.
+We globally set the `timeout connect` to be 5 seconds by default. (The default can be overridden as environment variable
+`ROUTER_DEFAULT_CONNECT_TIMEOUT` but only with a custom router.)  We also hard-code `timeout check` to 5 seconds and
+apply to the same backend scope as `check inter`.  As
+[documented in the HAProxy configuration manual](http://cbonte.github.io/haproxy-dconv/2.2/configuration.html#4.2-timeout%20check),
+when `timeout check` is set, HAProxy:
+>uses min(`timeout connect`, `inter`) as a connect timeout for check and `timeout check` as an additional read timeout
+
+This is likely to be a problem only if the `inter` duration was accidentally set to lower than 5 seconds, in which case
+the healthchecks could time out before the expected 5 seconds.  To mitigate this, the validation on the healthcheck
+interval value (`healthCheckInterval`) will have a minimum of 5 seconds.
+
+## Design Details
+
+### Open Questions [optional]
+
+Is there ever a reason to allow users to set the healthcheck interval lower than 5 seconds?
+
+### Test Plan
+
+Unit tests will be added to test the propagation of the `healthCheckInterval` setting and its interaction with the router
+annotation `router.openshift.io/haproxy.health.check.interval`.
+
+E2E tests can validate that the environment variable is present and the HAProxy template is properly constructed.
```

Comment:
I would think the unit test and E2E test would be swapped: a unit test in cluster-ingress-operator could verify that `desiredRouterDeployment` set the environment variable correctly, and an E2E test could verify that `router.openshift.io/haproxy.health.check.interval` took precedence over `healthCheckInterval`.  I might be misunderstanding this or looking at it the wrong way though, so if it makes sense to test as described here, that's fine as long as we have reasonable test coverage.